### PR TITLE
Feature/TMTabbedPage

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**

### CODE REVIEW

File: DemoApp/DemoApp/DemoApp.csproj [hunk 1]
```diff
@@ -70,20 +70,34 @@
...
 	  <None Remove="Resources\Images\search_icon.svg" />
+	  <None Remove="Views\TabViewItems\" />
 	</ItemGroup>
 
 	<ItemGroup>
 		<PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.0" />
 		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="7.0.0" />
 		<PackageReference Include="CommunityToolkit.Maui" Version="5.2.0" />
 		<PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
+		<PackageReference Include="Syncfusion.Maui.TabView" Version="22.2.7" />
```
Comment (rthanga1) [inline DemoApp/DemoApp/DemoApp.csproj:81]: Shouldn't be here.

File: DemoApp/DemoApp/Views/TMTabbedPageDemo.xaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+             xmlns:tabs="clr-namespace:DemoApp.Views.TabViewItems"
+             xmlns:constants="clr-namespace:DemoApp.Constant"
+             x:Class="DemoApp.Views.TMTabbedPageDemo">
+  <modus:TabViewItem Text="Home"
+                     Icon="{Static constants:ImageConstants.DarkGalleryIcon }">
+    <tabs:TabViewOne BackgroundColor="White"/>
+  </modus:TabViewItem>
+  <modus:TabViewItem Text="Mail"
+                     Icon="{Static constants:ImageConstants.Email }">
+    <tabs:TabViewTwo BackgroundColor="White"/>
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/TMTabbedPageDemo.xaml:17]: Same to all the views here

File: DemoApp/DemoApp/Views/TabViewItems/TabViewFour.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using DemoApp.ViewModels;
+
+namespace DemoApp.Views.TabViewItems;
+
+public partial class TabViewFour : ContentView
+{
+    public TabViewFour()
+    {
+        InitializeComponent();
+        BindingContext = new SegmentedControlViewModel();
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/TabViewItems/TabViewFour.xaml.cs:10]: BindingContext is not setting right. As a result, changing anything in the view doesn't change the appearance of the controls.
Screen Shot 2023-08-08 at 10 11 38 PM

From the logs:

> Microsoft.Maui.Controls.Xaml.Diagnostics.BindingDiagnostics: Warning: 'CurrentTextColor' property not found on 'Trimble.Modus.Components.TabViewItem', target property: 'Microsoft.Maui.Controls.Label.TextColor'
Microsoft.Maui.Controls.Xaml.Diagnostics.BindingDiagnostics: Warning: 'RadioButtons' property not found on 'DemoApp.ViewModels.SpinnerPageViewModel', target property: 'Trimble.Modus.Components.TMRadioButtonGroup.ItemsSource'
Microsoft.Maui.Controls.Xaml.Diagnostics.BindingDiagnostics: Warning: 'CurrentIcon' property not found on 'Trimble.Modus.Components.Model.SegmentItemModel', target property: 'Microsoft.Maui.Controls.Image.IsVisible'

File: Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+namespace Trimble.Modus.Components;
+
+[ContentProperty(nameof(TabItems))]
+public partial class TMTabbedPage : ContentPage
+{
+    #region Private Fields
+    private Grid mainContainer;
+    private Grid tabStripContainer;
+    private CarouselView contentContainer;
+    private List<double> contentWidthCollection;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:16]: What is the use of `contentWidthCollection`? Doesn't see to be used anywhere

File: Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+    private ObservableCollection<TabViewItem>? contentTabItems;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:17]: Why do we need to two arrays of TabViewItem?

File: Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+    #endregion
+    #region Public Fields
+    public ObservableCollection<TabViewItem> TabItems { get; set; } = new();
+    public delegate void TabSelectionChangedEventHandler(object? sender, TabSelectionChangedEventArgs e);
+    public event TabSelectionChangedEventHandler? SelectionChanged;
+    #endregion
+    #region Protected Fields
+    protected Grid _tabBarView = null;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:25]: Not used

Comment (rthanga1) [reply Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:25]: protected Grid _tabBarView = null;
    protected List<View> cells;
    protected List<View> selectedCells;

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Views/TMButtonPage.xaml [hunk 7]
```diff
@@ -4,9 +4,10 @@
              xmlns:constants="clr-namespace:DemoApp.Constant"
              xmlns:Modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
              x:Class="DemoApp.TMButtonPage"
+             BackgroundColor="Pink"
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TMButtonPage.xaml:7]: Removed

File: DemoApp/DemoApp/Views/MultiLineInputPage.xaml [hunk 8]
```diff
@@ -39,7 +39,8 @@
         </StackLayout>
 
         <Modus:MultiLineInput  x:Name="modusInput"
-                           Placeholder="Text Input"
+                           Placeholder="Text Input"
+                               HeightRequest="500"
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/MultiLineInputPage.xaml:43]: Removed

File: DemoApp/DemoApp/DemoApp.csproj (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/DemoApp.csproj:81]: Removed

File: DemoApp/DemoApp/Views/TMTabbedPageDemo.xaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+             xmlns:modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
+             xmlns:local="clr-namespace:DemoApp"
+             TabColor="Primary"
+             Orientation="Vertical"
+             xmlns:tabs="clr-namespace:DemoApp.Views.TabViewItems"
+             xmlns:constants="clr-namespace:DemoApp.Constant"
+             x:Class="DemoApp.Views.TMTabbedPageDemo">
+  <modus:TabViewItem Text="Home"
+                     Icon="{Static constants:ImageConstants.DarkGalleryIcon }">
+    <tabs:TabViewOne BackgroundColor="White"/>
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TMTabbedPageDemo.xaml:13]: Setting BackgroundColor is not required so removed

File: DemoApp/DemoApp/Views/TMTabbedPageDemo.xaml (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TMTabbedPageDemo.xaml:17]: Modified for all views

File: DemoApp/DemoApp/Views/TabViewItems/TabViewFour.xaml.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TabViewItems/TabViewFour.xaml.cs:10]: Set Binding Context as values are working now

File: Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:16]: Removed

File: Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:17]: the tab with contents are only added towards the strip which is contentTabItems, TabItems From the user contains all the tabs

File: Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs (hunk 6 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TabBar/TMTabbedPage.cs:25]: Removed

File: DemoApp/DemoApp/MainPage.xaml.cs [hunk 10]
```diff
@@ -65,8 +66,11 @@ private void ItemTapped(object sender, ItemTappedEventArgs e)
                     case "SegmentedControl":
                         Navigation.PushAsync(new TMSegmentedControlPage());
                         break;
-                    case "ListView":
+                    case "ListView":
                         Navigation.PushAsync(new TMListViewPage());
+                        break;
+                    case "TabbedPage":
+                        Navigation.PushAsync(new TMTabbedPageDemo());
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/MainPage.xaml.cs:73]: Removed

File: DemoApp/DemoApp/Views/TabViewItems/TabViewFour.xaml.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TabViewItems/TabViewFour.xaml.cs:10]: Tested after adding commands the context is accessible it works.

**Review by rthanga1 — DISMISSED** [review]

**Review by rthanga1 — APPROVED** [review]

**Review by BAL4-R — DISMISSED** [review]

**Review by rthanga1 — APPROVED** [review]

**Review by BAL4-R — APPROVED** [review]

**Pull request merged**